cli/auth.go

```diff
@@ -321,6 +325,129 @@ func (c *viamClient) createOrganizationAPIKey(orgID, keyName string) (*apppb.Cre
 	return c.client.CreateKey(c.c.Context, req)
 }
 
+// LocationAPIKeyCreateAction corresponds to `location api-key create`.
+func LocationAPIKeyCreateAction(cCtx *cli.Context) error {
+	c, err := newViamClient(cCtx)
+	if err != nil {
+		return err
+	}
+
+	err = c.locationAPIKeyCreateAction(cCtx)
+	return err
+}
+
+func (c *viamClient) locationAPIKeyCreateAction(cCtx *cli.Context) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+
+	locationID := cCtx.String(apiKeyFlagLocationID)
+	orgID := cCtx.String(apiKeyCreateFlagOrgID)
+	keyName := cCtx.String(apiKeyCreateFlagName)
+
+	if locationID == "" {
+		return errors.New("cannot create an api-key for a location without an ID")
+	}
+
+	if keyName == "" {
+		keyName = c.generateDefaultKeyName()
+		infof(cCtx.App.Writer, "using default key name of %s", keyName)
+	}
+
+	req := &apppb.CreateKeyRequest{
+		Name: keyName,
+		Authorizations: []*apppb.Authorization{
+			{
+				AuthorizationType: "role",
+				AuthorizationId:   "location_owner",
```

Comment:
[q] do we want to have an optional flag that could allow us to set "location_operator" and same for robot below? could still default to "owner" when omitted. doesn't look like we have that for organization though so maybe its better to have feature parity between the two to prevent confusion. may be something we add in the future though!

Reply:
yeah, we don't support diff roles right now. I think this is probably something we will want to add in the future.